tests/ui/needless_collect_indirect.stderr

```diff
@@ -47,9 +44,25 @@ LL | |     indirect_contains.contains(&&5);
    |
 help: Check if the original Iterator contains an element instead of collecting then checking
    |
-LL |     
 LL |     sample.iter().any(|x| x == &&5);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: avoid using `collect()` when not needed
+  --> $DIR/needless_collect_indirect.rs:21:5
+   |
+LL | /     let indirect_contains_with_comments = sample.iter().collect::<VecDeque<_>>();
+LL | |     // This is some stuff,
+LL | |     // to pretend that we have some more
+LL | |     // code present here
+...  |
+LL | |     // deal.
+LL | |     indirect_contains_with_comments.contains(&&5);
+   | |____^
+   |
+help: Check if the original Iterator contains an element instead of collecting then checking
    |
+LL |     sample.iter().any(|x| x == &&5);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
Wait, doesn't this remove the comment when auto-applied?

Reply:
Oops, this remove the comments. And I understood why this is using `multipart_suggestion` with `String::new()`.

Thank you for pointing that out! It seems difficult for me to improve this messeages, so I close this PR.